### DISCUSSION TITLE
terminal toggle

### DISCUSSION THREAD

**Discussion by jin-ahn** [LazyVim/LazyVim#193] [Q&A] [answered]
How do I make it so that I can hide float termainal pop and bring it back up? In its currently functinality it seems I have to quit the session and recreate it. I want to be able to continue a session

**Comment by alezunino** [comment, marked as answer]
Try adding this file:
~/.config/nvim/lua/plugins/toggleterm.lua
`return {
  {
    "akinsho/toggleterm.nvim",
    config = true,
    cmd = "ToggleTerm",
    keys = { { "<F4>", "<cmd>ToggleTerm<cr>", desc = "Toggle floating terminal" } },
    opts = {
      open_mapping = [[<F4>]],
      direction = "float",
      shade_filetypes = {},
      hide_numbers = true,
      insert_mappings = true,
      terminal_mappings = true,
      start_in_insert = true,
      close_on_exit = true,
    },
  },
}
`

**Reply by jin-ahn** [reply]
thank you!

**Reply by MaKaNu** [reply]
I tried this and several other approaches nothing worked. I use `:ToggleTerm` and it works. The key binding thing is difficult...

**Comment by felipe300** [comment]
how can i change the keybindings, i do it in this file or should i create another file?

**Reply by dpetka2001** [reply]
You should do it in the same file.

**Comment by andradei** [comment]
@felipe300  I recomment you keep the keybidings the same for `keys` and `open_mapping`. See line numbers 5 and 7 below 
![image](https://user-images.githubusercontent.com/2653546/221462864-e60af972-0288-4b08-bf1d-ecc046de0baa.png)

**Reply by dpetka2001** [reply]
I didn't ask the question. I just answered.

**Reply by andradei** [reply]
Indeed. Comment updated.

**Comment by rofrol** [comment]
```lua
  {
    "akinsho/toggleterm.nvim",
    config = function()
      require("toggleterm").setup({
        open_mapping = [[<c-\>]],
        shade_terminals = false,
        -- add --login so ~/.zprofile is loaded
        -- https://vi.stackexchange.com/questions/16019/neovim-terminal-not-reading-bash-profile/16021#16021
        shell = "zsh --login",
      })
    end,
    keys = {
      { [[<C-\>]] },
      { "<leader>0", "<Cmd>2ToggleTerm<Cr>", desc = "Terminal #2" },
      {
        "<leader>td",
        "<cmd>ToggleTerm size=40 dir=~/Desktop direction=horizontal<cr>",
        desc = "Open a horizontal terminal at the Desktop directory",
      },
    },
  },
```

`<Leader>0` open second terminal vertically.

- https://github.com/alpha2phi/modern-neovim/blob/268234975df52ef79237d248ad52a69590660e11/lua/plugins/toggleterm.lua#LL5C8-L5C8

**Reply by dpetka2001** [reply]
You might want to also take a look into something like this
```lua
      {
        "<leader>tv",
        function()
          local count = vim.v.count1
          require("toggleterm").toggle(count, 0, vim.loop.cwd(), "vertical")
        end,
        desc = "ToggleTerm (vertical)",
      },
      {
        "<leader>th",
        function()
          local count = vim.v.count1
          require("toggleterm").toggle(count, 10, vim.loop.cwd(), "horizontal")
        end,
        desc = "ToggleTerm (horizontal)",
      },
```
This way you just press a number and then the keybind and it will open `#2, #3, #4,etc` additional terminals either horizontally or vertically. Instead of relying on a specific keymap to pass the count for which terminal to open.

**Reply by hmada-godaddy** [reply]
can you give the example of how to use that config in the `/toggleterm.lua` file? the below cde doesn't work for me
`return {
  {
    "akinsho/toggleterm.nvim",
    config = true,
    cmd = "ToggleTerm",
    -- keys = { { "<C-t>", "<cmd>ToggleTerm<cr>", desc = "Toggle floating terminal" } },
    -- opts = {
    --   open_mapping = [[<C-t>]],
    --   direction = "horizontal",
    --   shade_filetypes = {},
    --   hide_numbers = true,
    --   insert_mappings = true,
    --   terminal_mappings = true,
    --   start_in_insert = true,
    --   close_on_exit = true,
    -- },

    {
      "<leader>tv",
      function()
        local count = vim.v.count1
        require("toggleterm").toggle(count, 0, vim.loop.cwd(), "vertical")
      end,
      desc = "ToggleTerm (vertical)",
    },
    {
      "<leader>th",
      function()
        local count = vim.v.count1
        require("toggleterm").toggle(count, 10, vim.loop.cwd(), "horizontal")
      end,
      desc = "ToggleTerm (horizontal)",
    },
  },
}`

**Reply by dpetka2001** [reply]
Take a look [here](https://github.com/dpetka2001/dotfiles/blob/main/dot_config/nvim/lua/plugins/toggleterm.lua) for `toggleterm`.

**Comment by rofrol** [comment]
[4.5.0](https://github.com/LazyVim/LazyVim/releases/tag/v4.5.0) addresses terminal toggle and shorcuts.

- `<Leader>ft` or `C-/` - toggle float terminal
- `<Leader>fT` - toggle CWD
- `c-h`, `c-j`, `c-k`, `c-l` - movement
- `c-w` - window mappings
- `c-ws` - split horizontally
- `c-wv` - split vertically
etc.

You still may want to bind `c-\` to toggle non-float terminal.
